wsgi_kerberos.py

```diff
@@ -127,57 +47,47 @@ class KerberosAuthMiddleware(object):
         is willing to read, the more vulnerable it becomes to denial-of-service
         attacks.
     :type read_max_on_auth_fail: int
+    :param logger: logging provider
+    :type logger: logging.Logger
     '''
-    def __init__(self, app, hostname='', unauthorized=None, forbidden=None,
-                 auth_required_callback=None, read_max_on_auth_fail=_DEFAULT_READ_MAX):
-        if hostname:
-            self._check_hostname(hostname)
-            self.service = 'HTTP@%s' % hostname
-        else:
-            self.service = ''
-        if unauthorized is None:
-            unauthorized = (b'Unauthorized', 'text/plain')
-        elif isinstance(unauthorized, basestring):
-            unauthorized = (unauthorized, 'text/plain')
-        unauthorized = (ensure_bytestring(unauthorized[0]), unauthorized[1])
-
-        if forbidden is None:
-            forbidden = (b'Forbidden', 'text/plain')
-        elif isinstance(forbidden, basestring):
-            forbidden = (forbidden, 'text/plain')
-        forbidden = (ensure_bytestring(forbidden[0]), forbidden[1])
-
-        if auth_required_callback is None:
-            auth_required_callback = lambda x: True
 
+    def __init__(
+        self,
+        app,
+        hostname='',
+        unauthorized=(b"Unauthorized", "text/plain"),
+        forbidden=(b"Forbidden", "text/plain"),
+        auth_required_callback=lambda _: True,
+        read_max_on_auth_fail=_DEFAULT_READ_MAX,
+        logger=_LOG,
+    ):
         self.application = app               # WSGI Application
         self.unauthorized = unauthorized     # 401 response text/content-type
         self.forbidden = forbidden           # 403 response text/content-type
         self.auth_required_callback = auth_required_callback
         self.read_max_on_auth_fail = read_max_on_auth_fail
+        self.logger = logger
 
-    @staticmethod
-    def _check_hostname(hostname):
-        try:
-            principal = kerberos.getServerPrincipalDetails('HTTP', hostname)
-        except kerberos.KrbError as exc:
-            LOG.warning('kerberos.getServerPrincipalDetails("HTTP", %r) raised %s', hostname, exc)
-        else:
-            LOG.debug('KerberosAuthMiddleware is identifying as %s', principal)
+        self.service = None
+        if hostname:
```

Comment:
Maybe default to `socket.gethostname()`?
```suggestion
        if not hostname:
            hostname = socket.gethostname()
```